openstack/antiddos/v1/antiddos/list_weekly_reports.go

```diff
@@ -0,0 +1,89 @@
+package antiddos
+
+import (
+	"encoding/json"
+	"strconv"
+	"time"
+
+	"github.com/opentelekomcloud/gophertelekomcloud"
+	"github.com/opentelekomcloud/gophertelekomcloud/internal/extract"
+)
+
+func ListWeeklyReports(client *golangsdk.ServiceClient, periodStartDate time.Time) (*ListWeeklyReportsResponse, error) {
+	// GET /v1/{project_id}/antiddos/weekly
+	raw, err := client.Get(
+		client.ServiceURL("antiddos", "weekly")+"?period_start_date="+strconv.FormatInt(periodStartDate.Unix()*1000, 10),
+		nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var res ListWeeklyReportsResponse
+	err = extract.Into(raw.Body, &res)
+	return &res, err
+}
+
+type ListWeeklyReportsResponse struct {
+	// Number of DDoS attacks intercepted in a week
+	DDosInterceptTimes int `json:"ddos_intercept_times"`
+	// Number of DDoS attacks intercepted in a week
+	Weekdata []WeekData `json:"-"`
+	// Top 10 attacked IP addresses
+	Top10 []struct {
+		// EIP
+		FloatingIpAddress string `json:"floating_ip_address"`
+		// Number of DDoS attacks intercepted, including cleaning operations and black-holes
+		Times int `json:"times"`
+	} `json:"top10"`
+}
+
+type WeekData struct {
+	// Number of DDoS attacks intercepted
+	DDosInterceptTimes int `json:"ddos_intercept_times"`
+	// Number of DDoS blackholes
+	DDosBlackholeTimes int `json:"ddos_blackhole_times"`
+	// Maximum attack traffic
+	MaxAttackBps int `json:"max_attack_bps"`
+	// Maximum number of attack connections
+	MaxAttackConns int `json:"max_attack_conns"`
+	// Start date
+	PeriodStartDate time.Time `json:"period_start_date"`
+}
+
+func (r *ListWeeklyReportsResponse) UnmarshalJSON(b []byte) error {
+	type tmp ListWeeklyReportsResponse
+	var s struct {
+		tmp
+		Weekdata []struct {
+			// Number of DDoS attacks intercepted
+			DDosInterceptTimes int `json:"ddos_intercept_times"`
+			// Number of DDoS blackholes
+			DDosBlackholeTimes int `json:"ddos_blackhole_times"`
+			// Maximum attack traffic
+			MaxAttackBps int `json:"max_attack_bps"`
+			// Maximum number of attack connections
+			MaxAttackConns int `json:"max_attack_conns"`
+			// Start date
+			PeriodStartDate int64 `json:"period_start_date"`
+		} `json:"weekdata"`
+	}
+	err := json.Unmarshal(b, &s)
+	if err != nil {
+		return err
+	}
+
+	*r = ListWeeklyReportsResponse(s.tmp)
```

Comment:
Why is it done like that? Is it some kind of special case? If yes - there should be a comment describing why there is an internal structure and direct JSON unmarshal instead of using SDK JSON handling.

Reply:
I don't actually see what the code is doing, I'm trying to figure out how to rewrite them.

Reply:
This is because PeriodStartDate has two different definitions: time.Time and int64.
This custom UnmarshalJSON performs the conversion between them.
This happens in other modules too, do we need it?

Reply:
Changing exported field type is not the greatest idea, at least without deprecation process. I would rather simplify `UnmarshalJSON` or move it to be a `Weekdata` method, converting json-mapped field (e.g. `PeriodStartTimestamp`) to the `time.Time` field with the old name